src/zenml/utils/io_utils.py

```diff
@@ -205,6 +205,31 @@ def resolve_relative_path(path: str) -> str:
     return str(Path(path).resolve())
 
 
+def is_safe_extraction_path(member_path: str, target_dir: str) -> bool:
```

Comment:
To make this more reusable, isn't this just a function that checks whether a path is inside another directory? The name could probably reflect that